### DISCUSSION TITLE
Why does my Github Pages service always show that SSL certificates are not allocated?

### DISCUSSION THREAD

**Discussion by ghost** [community/community#22790] [Other Feature Feedback, Questions, & Ideas]
<p>My Github Pages(<a>xiezheyuan.github.io</a>) has been redirected to the domain(<a>www.xiezheyuan.top</a>) I purchased from Aliyun and use CloudFlare to protection and acceleration my website.<br>
But Github show me:</p>
<blockquote>
<p>Certificate Request Error: Certificate provisioning will retry automatically in a short period, please be patient.<br>
What can I do?</p>
</blockquote>

**Comment by otter-computer** [comment, marked as answer]
<p>Hi <a class="mention" href="/xiezheyuan">@XieZheyuan</a>, welcome to the GitHub Support Community!</p>
<p>Cloudflare enables their HTTP/DNS proxy feature by default, meaning GitHub isn’t able to see the DNS records required to generate an HTTPS certificate. You’ll need to disable this for any DNS records that point towards GitHub.</p>
<p><strong>Disclaimer</strong> : Disabling the HTTP/DNS proxy may interfere with other Cloudflare services. If you use other Cloudflare services with your site, we aren’t able to guarantee that these will continue working. For more help with this, I’d recommend reaching out to Cloudflare’s support team.</p>
<p>If you’d still like to continue, you can follow the instructions below:</p>
<p>First of all, make sure you’ve set up your DNS records in Cloudflare according to the instructions in our help documentation:</p>
<aside class="onebox allowlistedgeneric">
  <header class="source">
      <img src="https://docs.github.com/assets/images/site/favicon.svg" class="site-icon" width="16" height="16">
      <a href="https://docs.github.com/en/free-pro-team@latest/github/working-with-github-pages/managing-a-custom-domain-for-your-github-pages-site" target="_blank" rel="noopener">docs.github.com</a>
  </header>
  <article class="onebox-body">
    <img src="" class="thumbnail" width="" height="">

<h3><a href="https://docs.github.com/en/free-pro-team@latest/github/working-with-github-pages/managing-a-custom-domain-for-your-github-pages-site" target="_blank" rel="noopener">Managing a custom domain for your GitHub Pages site - GitHub Docs</a></h3>



  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>To disable CloudFlare’s DNS/HTTP proxy, you’ll need to log in to your Cloudflare account, navigate to the DNS settings page for your domain, and click on the orange cloud icon next to any DNS records related to your GitHub Pages domain. The cloud icon should turn grey when it’s correctly disabled.</p>
<p>It can take up to 72 hours for these DNS changes to propagate across the internet. You can check the status of your DNS here:</p>
<aside class="onebox allowlistedgeneric">
  <header class="source">
      <img src="https://www.whatsmydns.net/favicon.ico" class="site-icon" width="48" height="48">
      <a href="https://www.whatsmydns.net/#CNAME/www.xiezheyuan.top" target="_blank" rel="noopener">whatsmydns.net</a>
  </header>
  <article class="onebox-body">
    <div class="aspect-image" style="--aspect-ratio:690/345;"><img src="https://user-images.githubusercontent.com/2439407/181073140-fa903018-57e0-4e19-99ff-efcf8aeaa875.png" class="thumbnail" width="690" height="345" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/5/59de7c2b434ea92fc358d2b6d4328e3decd69b66_2_690x345.png, https://aws1.discourse-cdn.com/github/optimized/2X/5/59de7c2b434ea92fc358d2b6d4328e3decd69b66_2_1035x517.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/5/59de7c2b434ea92fc358d2b6d4328e3decd69b66_2_1380x690.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/5/59de7c2b434ea92fc358d2b6d4328e3decd69b66_2_10x10.png"></div>

<h3><a href="https://www.whatsmydns.net/#CNAME/www.xiezheyuan.top" target="_blank" rel="noopener">DNS Checker - DNS Propagation Check &amp; DNS Lookup</a></h3>

<p>Instant DNS Propagation Check. Global DNS Lookup and DNS Propagation Checker. Check your DNS records around the world.</p>


  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>Once they do, you’ll need to trigger a recheck of your DNS setup by visiting your repository’s settings page. Simply visiting the page will kick off the process of provisioning an HTTPS certificate for your domain name.</p>
<p>Once your certificate has been provisioned, you’ll then be able to enforce HTTPS by checking the “Enforce HTTPS” checkbox.</p>
<p>Keep in mind that it can take an hour or so for an HTTPS certificate to be requested, so if you follow these steps but are still having trouble, that’s usually why.</p>
